sonar-plugin-api/src/main/java/org/sonar/api/SonarRuntime.java

```diff
@@ -111,80 +117,42 @@
  *      &lt;extensions&gt;true&lt;/extensions&gt;
  *      &lt;configuration&gt;
  *        &lt;!-- Override the default value 5.6 which is guessed from sonar-plugin-api dependency --&gt;
- *        &lt;sonarQubeMinVersion&gt;5.5&lt;/sonarQubeMinVersion&gt;
+ *        &lt;sonarQubeMinVersion&gt;5.6&lt;/sonarQubeMinVersion&gt;
  *      &lt;/configuration&gt;
  *    &lt;/plugin&gt;
  *  &lt;/plugins&gt;
  * &lt;/build&gt;
  * </pre>
  *
- *
+ * <p>
+ * Unit tests of plugin extensions can create instances of {@link SonarRuntime}
+ * via {@link org.sonar.api.internal.SonarRuntimeImpl}.
+ * </p>
  * @since 6.0
  */
 @ScannerSide
 @ServerSide
 @ComputeEngineSide
 @SonarLintSide
-@Immutable
-public class SonarRuntime {
+public interface SonarRuntime {
 
   /**
-   * Constant for version 5.5
-   */
-  public static final Version V5_5 = Version.create(5, 5);
-
-  /**
-   * Constant for version 5.6
-   */
-  public static final Version V5_6 = Version.create(5, 6);
-
-  /**
-   * Constant for version 6.0
-   */
-  public static final Version V6_0 = Version.create(6, 0);
-
-  private final Version version;
-  private final SonarProduct product;
-  private final SonarQubeSide sonarQubeSide;
-
-  public SonarRuntime(Version version, SonarProduct product, @Nullable SonarQubeSide sonarQubeSide) {
-    requireNonNull(version);
-    requireNonNull(product);
-    Preconditions.checkArgument((product == SonarProduct.SONARQUBE) == (sonarQubeSide != null), "sonarQubeSide should be provided only for SonarQube product");
-    this.version = version;
-    this.product = product;
-    this.sonarQubeSide = sonarQubeSide;
-  }
-
-  /**
-   * Runtime version of sonar-plugin-api. This could be used to test if a new feature can be used or not without using reflection.
-   */
-  public Version getApiVersion() {
-    return this.version;
-  }
-
-  public boolean isGreaterThanOrEqual(Version than) {
-    return this.version.isGreaterThanOrEqual(than);
-  }
+  * Version of API (sonar-plugin-api artifact) at runtime.
+  * It can be helpful to call some API classes/methods without checking their availability at
+  * runtime by using reflection.
+  */
+  Version getApiVersion();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Document this method return value. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUndocumentedApi)